### DISCUSSION TITLE
I'm looking for async/await API. Should I need them?

### DISCUSSION THREAD

**Discussion by chalermpong** [joel-perry/ApolloCombine#16] [Q&A] [answered]
Hi

I'm looking for async/await API. Should I need them?

Sometime I just want to simply call fetch/perform like this:
```swift
// For query
let result = await apollo.fetch(query: HeroNameQuery())
// For mutation
let result = await apollo.perform(mutation: UpvotePostMutation(postId: postId))
```

**Comment by joel-perry** [comment, marked as answer]
Hello!

Have a look at [this discussion](https://github.com/apollographql/apollo-ios/issues/2216) on the official Apollo iOS repo. It looks like there won't be official support for async/await until v2.0.

**Reply by chalermpong** [reply]
Thank you.
